From 059e631dac5dfd39a5ba1deb57ec89651f76b1ff Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Tue, 19 Aug 2014 15:27:02 -0400
Subject: [tools] perf/report: Use parse_options_usage() for -s option failure

Message-id: <1408462094-14194-254-git-send-email-jolsa@redhat.com>
Patchwork-id: 88212
O-Subject: [PATCH RHEL7.1 BZ1131394 253/325] perf report: Use parse_options_usage() for -s option failure
Bugzilla: 1131394
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1131394
https://bugzilla.redhat.com/show_bug.cgi?id=1131394

upstream
========
commit 91aba0a62e24ff7a567e13e1d88deab711df6f0f
Author: Namhyung Kim <namhyung.kim@lge.com>
Date: Fri Nov 1 16:33:13 2013 +0900

description
===========
The -s (--sort) option was processed after normal option parsing so that
it cannot call the parse_options_usage() automatically.  Currently it
calls usage_with_options() which shows entire help messages for event
option.  Fix it by showing just -s options.

  $ perf report -s help
    Error: Unknown --sort key: `help'

   usage: perf report [<options>]

      -s, --sort <key[,key2...]>
                            sort by key(s): pid, comm, dso, symbol, ...
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/builtin-report.c b/tools/perf/builtin-report.c
index 4df3161..25f83d5 100644
--- a/tools/perf/builtin-report.c
+++ b/tools/perf/builtin-report.c
@@ -947,8 +947,10 @@ repeat:
    sort_order = "local_weight,mem,sym,dso,symbol_daddr,dso_daddr,snoop,tlb,locked";
  }
 
- if (setup_sorting() < 0)
-  usage_with_options(report_usage, options);
+ if (setup_sorting() < 0) {
+  parse_options_usage(report_usage, options, "s", 1);
+  goto error;
+ }
 
  if (parent_pattern != default_parent_pattern) {
   if (sort_dimension__add("parent") < 0)
-- 
1.7.1